### PR TITLE
Allow skipping closure-compiler

### PR DESCRIPTION
**Pull request by tomsmalley**

### CODE REVIEW

**Comment by ryantrinkle** [comment]
Is this something we want to be configured at the project level?  I can see a use-case there, e.g. as an escape hatch when something isn't compatible with closure compiler, but it seems like we might also want a way to build without closure compiler in a one-off way, e.g. with a command line argument to nix-build or something.  Is that possible with this approach?

**Comment by tomsmalley** [comment]
No, but I can add it to the skeleton's default.nix args I suppose

File: default.nix [hunk 1]
```diff
@@ -164,12 +164,18 @@ in rec {
...
+  } ''
     mkdir $out
     cd $out
     ln -s "${haskellLib.justStaticExecutables frontend}/bin/frontend.jsexe/all.js" all.unminified.js
-    closure-compiler --externs "${reflex-platform.ghcjsExternsJs}" -O ${optimizationLevel} --jscomp_warning=checkVars --create_source_map="all.js.map" --source_map_format=V3 --js_output_file="all.js" all.unminified.js
-    echo "//# sourceMappingURL=all.js.map" >> all.js
+    ${if optimizationLevel == null then ''
+      ln -s all.unminified.js all.js
+    '' else ''
+      closure-compiler --externs "${reflex-platform.ghcjsExternsJs}" -O ${optimizationLevel} --jscomp_warning=checkVars --create_source_map="all.js.map" --source_map_format=V3 --js_output_file="all.js" all.unminified.js
```
Comment (3noch) [inline default.nix]: Nit: Using `${pkgs.closurecompiler}/bin/closure-compiler` here would avoid having to fiddle with `buildInputs` above.

**Comment by 3noch** [comment]
As far as I can tell, this design makes it difficult to deploy uncompressed JS. In order to get uncompressed you need to `nix-build -arg projectOverrides '{ __closureCompilerOptimizationLevel = null }'`. However `ob deploy` won't pass this argument to `nix-build` and doesn't have a way to let you do so. The only way to get around this would be to change the default in `default.nix` directly which would mean your deployment would match your defaults anyway.

**Comment by 3noch** [comment]
It's probably worth noting, however, that "dev" usually means `ob run` or similar which doesn't even use GHCJS, so there's already a pretty big leap from that to production. It would be nice to support a more production-like development setup as #302 requests.

File: default.nix (hunk 1 above)
Comment (tomsmalley) [reply default.nix]: Nice catch :+1:

**Comment by mightybyte** [comment]
I would also like to be able to disable closure compiling on a one-off basis to speed up the situations where I need to test with the generated javascript.

**Comment by tomsmalley** [comment]
@ali-abrar resolved

**Pull request merged**